tests/test_dom_modify.cpp

```diff
@@ -1762,6 +1762,19 @@ TEST(dom_fp_roundtrip_min_max_double)
 	CHECK(fp_equal(node.text().as_double(), std::numeric_limits<double>::max()));
 }
 
+TEST(dom_fp_double_custom_precision)
+{
+	xml_document doc;
+	xml_node node = doc.append_child(STR("node"));
+	xml_attribute attr = node.append_attribute(STR("attr"));
+
+	attr.set_value(std::numeric_limits<double>::min(), 20);
```

Comment:
Would be great to have a couple of tests that just test basic float values, e.g. set_value(1.0f, 5) and set_value(3.1415926f, 3).